core/sr-io/without_std.rs

```diff
@@ -331,8 +331,102 @@ pub mod ext {
 		// Offchain-worker Context
 		//================================
 
-		/// Submit extrinsic.
-		fn ext_submit_extrinsic(data: *const u8, len: u32);
+		/// Submit transaction.
+		///
+		/// Returns 0 if it was successfuly added to the pool, nonzero otherwise.
+		fn ext_submit_transaction(data: *const u8, len: u32) -> u32;
+
+		/// Sign a piece of data with authority key.
+		///
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_sign(data: *const u8, len: u32, sig_data: *mut u8) -> u32;
+
+		/// Returns current UNIX timestamp (milliseconds)
+		fn ext_timestamp() -> u64;
+
+		/// Pause execution until `deadline` is reached.
+		fn ext_sleep_until(deadline: u64);
+
+		/// Generate a random seed
+		fn ext_random_seed(data: *mut u8);
+
+		/// Write a value to local storage.
+		fn ext_local_storage_set(key: *const u8, key_len: u32, value: *const u8, value_len: u32);
+
+		/// Read a value from local storage.
+		///
+		/// if len is u32::max_value the value has not been found.
+		fn ext_local_storage_read(key: *const u8, key_len: u32, value_len: *mut u32) -> *mut u8;
+
+		/// Initiaties a http request.
+		///
+		/// If returned value is `u32::max_value` the request couldn't have started.
+		fn ext_http_request_start(
+			method: *const u8,
+			method_len: u8,
+			url: *const u8,
+			url_len: u32,
+			meta: *const u8,
+			meta_len: u32
+		) -> u32;
+
+		/// Add a header to the request.
+		///
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_http_request_add_header(
+			request_id: u32,
+			name: *const u8,
+			name_len: u32,
+			value: *const u8,
+			value_len: u32
+		) -> u32;
+
+		/// Write a chunk of request body.
+		///
+		/// Writing an empty chunks finalises the request.
+		/// Passing `0` as deadline blocks forever.
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_http_request_write_body(
+			request_id: u32,
+			chunk: *const u8,
+			chunk_len: u32,
+			deadline: u64
+		) -> u32;
+
+		/// Block and wait for the responses for given requests.
+		///
+		/// Note that if deadline is not provided the method will block indefinitely,
+		/// otherwise unready responses will produce `WaitTimeout` status.
+		///
+		/// Make sure that `statuses` have the same length as ids.
+		/// Passing `0` as deadline blocks forever.
+		fn ext_http_response_wait(
+			ids: *const u32,
+			ids_len: u32,
+			statuses: *mut u32,
+			deadline: u64
+		);
+
+		/// Read all response headers.
+		///
+		/// Resturns parity-codec encoded vector of pairs `(HeaderKey, HeaderValue)`.
```

Comment:
```suggestion
		/// Returns parity-codec encoded vector of pairs `(HeaderKey, HeaderValue)`.
```

Reply:
I am a bit concerned that it doesn't return a `Result` and at the same time doesn't mention any preconditions like we should call this method only if the status of `id` is appropriate.